### PR TITLE
Enrich request/response struct

### LINKED ISSUE #3919: Inconsistency when initializing HTTP Response and Error
Consider the following code segment,

```
http:Response clientResponse = {};
http:HttpConnectorError err = {};
println(err);
println(clientResponse);
```
Above gives following output.

```
{msg:"", cause:null, stackTrace:null, statusCode:0}
{}
```

### LINKED ISSUE #4124: [HTTP] Enrich both request and response structs
At the moment above structs are opaque to ballerina developer. We need to make it a bit more interesting by adding headers, uri, host/port information to these structs.

### PR DESCRIPTION
**Pull request by chamil321**
PR includes property and header fields to request/response struct in the following manner.
```ballerina
struct Request {
   string remoteHost;
   int port;
   string path;
   string method;
   string httpVersion;
   string userAgent;
   map headers;
}

struct Response {
    int statusCode;
    string reasonPhrase;
    string server;
    map headers;
}
```
The headers map consists of string header key and HeaderValue struct array
```ballerina
public struct HeaderValue {
    string value;
    map param;
}
```

- Added utility methods for header such as getHeader, getHeaders, setHeader, addHeader, removeHeader and removeAllHeaders in ballerina.
- Fixes #4124 , Fixes #3919
- ATM struct field values can be changed. But in the next iteration, inbound req/resp struct fields should be read-only.

### CODE REVIEW

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 1]
```diff
@@ -2,6 +2,19 @@ package ballerina.net.http;
 
 @Description { value:"Represents an HTTP request message"}
 public struct Request {
+    string path;
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Indentation issue

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 2]
```diff
@@ -2,6 +2,19 @@ package ballerina.net.http;
 
 @Description { value:"Represents an HTTP request message"}
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Add @Field  annotations with proper details for each struct field

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 3]
```diff
@@ -2,6 +2,19 @@ package ballerina.net.http;
...
+    string path;
+	string host;
+	int port;
+	string method;
+	string httpVersion;
+	string userAgent;
+	map headers;
+}
+
+@Description { value:"Represents an HTTP header value"}
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Add @Field doc annotations.

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 4]
```diff
@@ -121,6 +134,10 @@ public native function <Request req> setHeader (string key, string value);
 
 @Description { value:"Represents an HTTP response message"}
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Add @Field doc annotations.

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 5]
```diff
@@ -121,6 +134,10 @@ public native function <Request req> setHeader (string key, string value);
...
 public struct Response {
+    int statusCode;
+    string reasonPhrase;
+	string server;
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Indentation issue

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal (hunk 2 above)
Comment (chamil321) [reply modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Ack

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal (hunk 3 above)
Comment (chamil321) [reply modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Ack

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal (hunk 4 above)
Comment (chamil321) [reply modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Ack

**Comment by chamil321** [comment]
I renamed files and ATM it includes functions which are implemented in ballerina.

**Pull request merged**